pjlib/src/pj/ioqueue_epoll.c

```diff
@@ -578,20 +586,11 @@ static void ioqueue_remove_from_set( pj_ioqueue_t *ioqueue,
                                      pj_ioqueue_key_t *key, 
                                      enum ioqueue_event_type event_type)
 {
-#if USE_EPOLLONESHOT
```

Comment:
Now `ioqueue_remove_from_set()` will only executed when event_type is `WRITEABLE_EVENT`.
The method itself is called from `ioqueue_dispatch_read_event(), ioqueue_dispatch_exception_event() and ioqueue_dispatch_write_event()`.
On read and exception event, then there's no rearm? And this patch will call `ioqueue_add_to_set()` only when there's something wrong on the dispatch method.

Reply:
> Now `ioqueue_remove_from_set()` will only executed when event_type is `WRITEABLE_EVENT`.

Correct, because `remove_from_set()` is supposed to remove the event, instead of adding/rearming it, which can cause premature or unexpected rearming.

> The method itself is called from `ioqueue_dispatch_read_event(), ioqueue_dispatch_exception_event() and ioqueue_dispatch_write_event()`.
> On read and exception event, then there's no rearm? And this patch will call `ioqueue_add_to_set()` only when there's something wrong on the dispatch method.

For success case of dispatch, the rearm will be done when `ioqueue_add_to_set()` is called.
For example: `ioqueue_dispatch_read_event()` -> `on_read_complete()` -> `pj_ioqueue_recv/recvfrom()` -> `ioqueue_add_to_set()`.

Reply:
I think there's a possibility that `ioqueue_add_to_set()` won't be called after the dispatch event.
e.g: on `pj_ioqueue_recv()` [ref](https://github.com/pjsip/pjproject/blob/master/pjlib/src/pj/ioqueue_common_abs.c#L767)
There's a possibility the method returns without calling `ioqueue_add_to_set()`. [ref](https://github.com/pjsip/pjproject/blob/master/pjlib/src/pj/ioqueue_common_abs.c#L736).
I think we should re-arm on that scenario as well.

Reply:
Good catch!

On normal scenario, it should be okay, since immediate read doesn't need event notification, and normally app will continue reading until no further data and the read status is pending/async (such as in [sip_transport_udp](https://github.com/pjsip/pjproject/blob/master/pjsip/src/pjsip/sip_transport_udp.c#L169) or pjmedia/transport_udp).

But with oneshot, it can be quite fragile indeed since it's difficult to track every flow and any error may cause the rearm not to happen. So to be safe and more robust, I modify it as in the next commit.